### DISCUSSION TITLE
Is there any way to get non valid Terraform files updated by Renovate

### DISCUSSION THREAD

**Discussion by menzbua** [renovatebot/renovate#22097] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.79.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Hi,

I have a problem getting non-valid Terraform files updated by Renovate. We have some repos with templated Terraform files. That means that the files are not Terraform conform in the repo. They would be deployed by Backstage, and then the templating would generate valid Terraform files. Here an example of such a file:
```tf
module "my-terraform-module" {
  source                 = "our-local-terraform-module-registry"
  version                = "0.3.0"
  name                   = "${{ values.name }}"
  description            = "${{ values.name }}"
 location    = var.location
  lan_name   = "${{ values.name }}"
  lan_public = false
}
```
Renovate would find all the Terraform files, but only files without the templated values would be processed:
```
DEBUG: Matched 15 file(s) for manager terraform: config-template/provision/backend.tf, config-template/provision/main.tf, config-template/provision/variables.tf, provision/backend.tf, provision/main.tf, provision/output.tf, provision/provider.tf, provision/user.tf, provision/variables.tf, template/provision/backend.tf, template/provision/main.tf, template/provision/output.tf, template/provision/provider.tf, template/provision/user.tf, template/provision/variables.tf
DEBUG: manager extract durations (ms)
       "managers": {"terraform": 49}
DEBUG: Found terraform package files
DEBUG: Found 5 package file(s)
 INFO: Dependency extraction complete
       "stats": {
         "managers": {
           "terraform": {"fileCount": 3, "depCount": 3}
         },
         "total": {"fileCount": 3, "depCount": 3}
       }

```
Is there any way to ignore the non-valid values in a Terraform file and do the upgrades anyway by Renovate?
Thanks and Regards,
Manuel

### What you tried so far.

I don't know what i can try to solve this issue. That's why I'm asking.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
For now, your only possibility is to define a regex manager

**Reply by menzbua** [reply]
Thanks @rarkins. Then i have to cover it with a regex.
